### ISSUE TITLE
NNW Safari extension does not recognise feed

### ISSUE THREAD

**Issue by Thomas-U** [brentsimmons/NetNewsWire#1348]
From time to time I encounter a webpage where the NNW-extension is light grey, indicating that it can not see any feed. When I add the URL manually in NNW it finds the feed instantly.
Latest example is Tim Berners-Lee’s new project: https://contractfortheweb.org/ . Manually the feed is recognised both from home and blog page URL. The extension sees neither.

**Comment by vincode-io** [comment]
@Thomas-U can you get a screenshot next time this happens?  I’ve never seen it before.

**Comment by vincode-io** [comment]
Never mind @Thomas-U. I see now that this is for the Mac extension.

**Comment by Thomas-U** [comment]
another page with the same issue:

https://bombich.com/blog

**Comment by danielpunkass** [comment]
Neither https://bombich.com/blog nor https://contractfortheweb.org advertises a "link" with a feed-type tag. This is currently what the browser extension is limited to as far as detecting feeds. The app must be doing something more extensive, probably scraping the page content for likely HTML links to a feed. I'll see what I can do to get migrate as much of the app logic as I can over to the extension.

**Comment by brentsimmons** [comment]
The app does go through and check page content to find a feed. It might be doing more stuff than a browser extension should (I’m not sure how much work is appropriate for a browser extension).

**Comment by danielpunkass** [comment]
Yeah, it probably comes down to whether the browser extension can do a bit more than it does, without too much cost. Since it has to perform its search on every single page load, whereas the app only does it in response to user's implicit request to look for a feed.

**Comment by Thomas-U** [comment]
@brentsimmons: the bombich page has an RSS button with the link "https://bombich.com/blog/feed" on the right sidebar.

**Comment by Thomas-U** [comment]
@danielpunkass Can you please give me feedback if you want more examples?

Today I saw that HuffPost has a RSS-link in the footer but the NNW icon stays light grey:
https://www.huffpost.com

**Comment by danielpunkass** [comment]
Thanks @Thomas-U  - probably don't need more examples unless you identify something that specfically has an RSS link the <head> section of the page source, which is the only place the Safari plugin currently looks.

**Comment by Wevah** [comment]
The RSS link in HuffPost's footer goes to another page that links to all the feeds (ending in `/feed`), so that wouldn't be a good link to detect anyway.

Could do detection for URL path ends with `/feed`, but that could be a deep rabbit hole to even more heuristics.

**Comment by brentsimmons** [comment]
Fixed in 7.1.3 as much as possible. The extension now looks for possible feeds inside page content. It doesn't have a network entitlement, so it's not going to go out and actually verify — instead, the best guess at a feed URL gets sent to NetNewsWire, which will make network requests.

Unfortunately, this doesn't deal with https://contractfortheweb.org/ which has no feed hints of any kind, it appears.

**Comment by brentsimmons** [comment]
On second thought — I’m going to try just always enabling the button so it can send to NetNewsWire, and NetNewsWire can then find the feed.
